Compile error with VK_ENABLE_BETA_EXTENSIONS

I'm seeing new compile errors with the 153 headers when including vulkan.hpp with `VK_ENABLE_BETA_EXTENSIONS` defined:
```
#define VK_ENABLE_BETA_EXTENSIONS
#include <vulkan/vulkan.hpp>
```
g++ output:
```
In file included from Vulkan-Headers/include/vulkan/vulkan.hpp:6794,
                 from hpptest.cpp:2:
Vulkan-Headers/include/vulkan/vulkan_handles.hpp:12039:114: error: ‘StructureChainAllocator’ does not name a type; did you mean ‘StructureChainValidation’?
12039 |            typename B2                                                                                      = StructureChainAllocator,
      |                                                                                                               ^~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                                               StructureChainValidation
Vulkan-Headers/include/vulkan/vulkan_handles.hpp:12040:91: error: type/value mismatch at argument 2 in template parameter list for ‘template<class, class> struct std::is_same’
12040 |               typename std::enable_if<std::is_same<typename B2::value_type, StructureChain>::value, int>::type = 0>
      |                                                                                           ^
Vulkan-Headers/include/vulkan/vulkan_handles.hpp:12040:91: note:   expected a type, got ‘StructureChain’
Vulkan-Headers/include/vulkan/vulkan_handles.hpp:12040:104: error: template argument 1 is invalid
12040 |               typename std::enable_if<std::is_same<typename B2::value_type, StructureChain>::value, int>::type = 0>
      |                                                                                                        ^
Vulkan-Headers/include/vulkan/vulkan_handles.hpp:12040:105: error: ‘<declaration error>’ is not a template [-fpermissive]
12040 |               typename std::enable_if<std::is_same<typename B2::value_type, StructureChain>::value, int>::type = 0>
      |                                                                                                         ^~
Vulkan-Headers/include/vulkan/vulkan_handles.hpp:12043:44: error: ‘StructureChainAllocator’ has not been declared
12043 |                                            StructureChainAllocator &                                            structureChainAllocator,
      |                                            ^~~~~~~~~~~~~~~~~~~~~~~
In file included from Vulkan-Headers/include/vulkan/vulkan.hpp:6796:
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp: In member function ‘typename vk::ResultValueType<std::pair<vk::StructureChain<X, Y, Z ...>, std::vector<unsigned char, Uint8_tAllocator> > >::type vk::Device::getEncodedVideoSessionParametersKHR(const vk::VideoEncodeSessionParametersGetInfoKHR&, const Dispatch&) const’:
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp:18662:9: error: ‘structureChains’ was not declared in this scope; did you mean ‘StructureChain’?
18662 |         structureChains.resize( dataSize );
      |         ^~~~~~~~~~~~~~~
      |         StructureChain
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp:18666:59: error: expected primary-expression before ‘void’
18666 |           data[i].pNext = structureChains[i].template get<void>().pNext;
      |                                                           ^~~~
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp: At global scope:
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp:18686:89: error: type/value mismatch at argument 2 in template parameter list for ‘template<class, class> struct std::is_same’
18686 |             typename std::enable_if<std::is_same<typename B2::value_type, StructureChain>::value, int>::type>
      |                                                                                         ^
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp:18686:89: note:   expected a type, got ‘StructureChain’
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp:18686:102: error: template argument 1 is invalid
18686 |             typename std::enable_if<std::is_same<typename B2::value_type, StructureChain>::value, int>::type>
      |                                                                                                      ^
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp:18686:103: error: ‘<declaration error>’ is not a template [-fpermissive]
18686 |             typename std::enable_if<std::is_same<typename B2::value_type, StructureChain>::value, int>::type>
      |                                                                                                       ^~
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp:18690:50: error: ‘StructureChainAllocator’ has not been declared
18690 |                                                  StructureChainAllocator &                                            structureChainAllocator,
      |                                                  ^~~~~~~~~~~~~~~~~~~~~~~
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp: In member function ‘typename vk::ResultValueType<std::pair<vk::StructureChain<X, Y, Z ...>, std::vector<unsigned char, Uint8_tAllocator> > >::type vk::Device::getEncodedVideoSessionParametersKHR(const vk::VideoEncodeSessionParametersGetInfoKHR&, int&, const Dispatch&) const’:
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp:18696:81: error: ‘uint8_tAllocator’ was not declared in this scope; did you mean ‘Uint8_tAllocator’?
18696 |       std::piecewise_construct, std::forward_as_tuple(), std::forward_as_tuple( uint8_tAllocator ) );
      |                                                                                 ^~~~~~~~~~~~~~~~
      |                                                                                 Uint8_tAllocator
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp:18711:9: error: ‘structureChains’ was not declared in this scope; did you mean ‘StructureChain’?
18711 |         structureChains.resize( dataSize );
      |         ^~~~~~~~~~~~~~~
      |         StructureChain
Vulkan-Headers/include/vulkan/vulkan_funcs.hpp:18715:59: error: expected primary-expression before ‘void’
18715 |           data[i].pNext = structureChains[i].template get<void>().pNext;
      |                                                           ^~~~

```

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -2234,22 +2234,7 @@ std::string VulkanHppGenerator::generateArgumentListEnhanced( std::vector<ParamD
   }
   if ( withAllocators )
   {
-    bool useStructureChainAllocator = false;
-    if ( flavourFlags & CommandFlavourFlagBits::chained )
-    {
-      // use StructureChainAllocator only, if a vector param is a chained return param
-      auto it = std::find_if(
-        chainedReturnParams.begin(), chainedReturnParams.end(), [&vectorParams]( size_t crp ) { return vectorParams.find( crp ) != vectorParams.end(); } );
-      if ( it != chainedReturnParams.end() )
-      {
-        // assert that there's no other chained vector param !
-        assert( std::find_if( std::next( it ),
-                              chainedReturnParams.end(),
-                              [&vectorParams]( size_t crp ) { return vectorParams.find( crp ) != vectorParams.end(); } ) == chainedReturnParams.end() );
-        useStructureChainAllocator = true;
-      }
-    }
-    if ( useStructureChainAllocator )
+    if ( ( flavourFlags & CommandFlavourFlagBits::chained ) && needsStructureChainResize( vectorParams, chainedReturnParams ) )
     {
       if ( encounteredArgument )
       {
@@ -2928,6 +2913,7 @@ std::string VulkanHppGenerator::generateCallSequence( std::string const &       
                                                       size_t                                    initialSkipCount,
                                                       std::set<size_t> const &                  singularParams,
                                                       std::set<size_t> const &                  templatedParams,
+                                                      std::vector<size_t> const &               chainedReturnParams,
                                                       CommandFlavourFlags                       flavourFlags,
                                                       bool                                      raii ) const
 {
@@ -2951,7 +2937,7 @@ std::string VulkanHppGenerator::generateCallSequence( std::string const &       
     std::string vectorName          = startLowerCase( stripPrefix( commandData.params[vectorParamIt->first].name, "p" ) );
     std::string vectorSize          = startLowerCase( stripPrefix( commandData.params[vectorParamIt->second.lenParam].name, "p" ) );
 
-    if ( flavourFlags & CommandFlavourFlagBits::chained )
+    if ( ( flavourFlags & CommandFlavourFlagBits::chained ) && needsStructureChainResize( vectorParams, chainedReturnParams ) )
     {
       assert( vectorParams.size() == 1 );
       // chained data needs some more handling!!
@@ -3282,8 +3268,8 @@ std::string VulkanHppGenerator::generateCommandEnhanced( std::string const &    
     std::string dataPreparation =
       generateDataPreparation( commandData, initialSkipCount, returnParams, vectorParams, templatedParams, flavourFlags, enumerating );
     std::string dataSizeChecks = generateDataSizeChecks( commandData, returnParams, dataTypes, vectorParams, templatedParams, singular );
-    std::string callSequence =
-      generateCallSequence( name, commandData, returnParams, vectorParams, initialSkipCount, singularParams, templatedParams, flavourFlags, false );
+    std::string callSequence   = generateCallSequence(
+      name, commandData, returnParams, vectorParams, initialSkipCount, singularParams, templatedParams, chainedReturnParams, flavourFlags, false );
     std::string resultCheck     = generateResultCheck( commandData, className, classSeparator, commandName, enumerating );
     std::string returnStatement = generateReturnStatement( name,
                                                            commandData,
@@ -7391,8 +7377,8 @@ ${vectorSizeCheck}
   }
 )";
 
-    std::string callSequence =
-      generateCallSequence( name, commandData, returnParams, vectorParams, initialSkipCount, singularParams, templatedParams, flavourFlags, true );
+    std::string callSequence = generateCallSequence(
+      name, commandData, returnParams, vectorParams, initialSkipCount, singularParams, templatedParams, chainedReturnParams, flavourFlags, true );
     std::string className      = initialSkipCount ? stripPrefix( commandData.params[initialSkipCount - 1].type.type, "Vk" ) : "Context";
     std::string returnVariable = generateReturnVariable( commandData, returnParams, vectorParams, flavourFlags );
     std::string dataDeclarations =
@@ -11443,6 +11429,19 @@ bool VulkanHppGenerator::isTypeUsed( std::string const & type ) const
     }
   }
   return false;
+}
+
+bool VulkanHppGenerator::needsStructureChainResize( std::map<size_t, VectorParamData> const & vectorParams,
+                                                    std::vector<size_t> const &               chainedReturnParams ) const
+{
+  auto it = std::find_if(
+    chainedReturnParams.begin(), chainedReturnParams.end(), [&vectorParams]( size_t crp ) { return vectorParams.find( crp ) != vectorParams.end(); } );
+  // assert that there's no other chained vector param !
+  assert(
+    ( it == chainedReturnParams.end() ) ||
+    ( std::find_if( std::next( it ), chainedReturnParams.end(), [&vectorParams]( size_t crp ) { return vectorParams.find( crp ) != vectorParams.end(); } ) ==
+      chainedReturnParams.end() ) );
+  return ( it != chainedReturnParams.end() );
 }
 
 std::pair<bool, std::map<size_t, std::vector<size_t>>> VulkanHppGenerator::needsVectorSizeCheck( std::vector<ParamData> const &            params,

--- a/VulkanHppGenerator.hpp
+++ b/VulkanHppGenerator.hpp
@@ -540,6 +540,7 @@ private:
                                     size_t                                    initialSkipCount,
                                     std::set<size_t> const &                  singularParams,
                                     std::set<size_t> const &                  templatedParams,
+                                    std::vector<size_t> const &               chainedReturnParams,
                                     CommandFlavourFlags                       flavourFlags,
                                     bool                                      raii ) const;
   std::string generateChainTemplates( std::vector<size_t> const & returnParams, bool chained ) const;
@@ -936,6 +937,7 @@ private:
   bool isSupportedFeature( std::string const & name ) const;
   bool isTypeRequired( std::string const & type ) const;
   bool isTypeUsed( std::string const & type ) const;
+  bool needsStructureChainResize( std::map<size_t, VectorParamData> const & vectorParams, std::vector<size_t> const & chainedReturnParams ) const;
   std::pair<bool, std::map<size_t, std::vector<size_t>>> needsVectorSizeCheck( std::vector<ParamData> const &            params,
                                                                                std::map<size_t, VectorParamData> const & vectorParams,
                                                                                std::vector<size_t> const &               returnParams,

--- a/tests/EnableBetaExtensions/EnableBetaExtensions.cpp
+++ b/tests/EnableBetaExtensions/EnableBetaExtensions.cpp
@@ -18,7 +18,13 @@
 #define VK_ENABLE_BETA_EXTENSIONS
 #include <vulkan/vulkan.hpp>
 
-int main(int /*argc*/, char** /*argv*/)
+int main( int /*argc*/, char ** /*argv*/ )
 {
+  vk::Device device;
+
+  vk::VideoEncodeSessionParametersGetInfoKHR videoSessionParametersInfo;
+  auto                                       stuff = device.getEncodedVideoSessionParametersKHR<vk::VideoEncodeSessionParametersFeedbackInfoKHR,
+                                                          vk::VideoEncodeH264SessionParametersFeedbackInfoEXT,
+                                                          vk::VideoEncodeH265SessionParametersFeedbackInfoEXT>( videoSessionParametersInfo );
   return 0;
 }

--- a/vulkan/vulkan_funcs.hpp
+++ b/vulkan/vulkan_funcs.hpp
@@ -18660,12 +18660,7 @@ namespace VULKAN_HPP_NAMESPACE
                                                         nullptr );
       if ( ( result == VK_SUCCESS ) && dataSize )
       {
-        structureChains.resize( dataSize );
         data.resize( dataSize );
-        for ( size_t i = 0; i < dataSize; i++ )
-        {
-          data[i].pNext = structureChains[i].template get<void>().pNext;
-        }
         result = d.vkGetEncodedVideoSessionParametersKHR( m_device,
                                                           reinterpret_cast<const VkVideoEncodeSessionParametersGetInfoKHR *>( &videoSessionParametersInfo ),
                                                           reinterpret_cast<VkVideoEncodeSessionParametersFeedbackInfoKHR *>( &feedbackInfo ),
@@ -18709,12 +18704,7 @@ namespace VULKAN_HPP_NAMESPACE
                                                         nullptr );
       if ( ( result == VK_SUCCESS ) && dataSize )
       {
-        structureChains.resize( dataSize );
         data.resize( dataSize );
-        for ( size_t i = 0; i < dataSize; i++ )
-        {
-          data[i].pNext = structureChains[i].template get<void>().pNext;
-        }
         result = d.vkGetEncodedVideoSessionParametersKHR( m_device,
                                                           reinterpret_cast<const VkVideoEncodeSessionParametersGetInfoKHR *>( &videoSessionParametersInfo ),
                                                           reinterpret_cast<VkVideoEncodeSessionParametersFeedbackInfoKHR *>( &feedbackInfo ),

--- a/vulkan/vulkan_raii.hpp
+++ b/vulkan/vulkan_raii.hpp
@@ -18567,12 +18567,7 @@ namespace VULKAN_HPP_NAMESPACE
           nullptr );
         if ( ( result == VK_SUCCESS ) && dataSize )
         {
-          structureChains.resize( dataSize );
           data.resize( dataSize );
-          for ( size_t i = 0; i < dataSize; i++ )
-          {
-            data[i].pNext = structureChains[i].template get<void>().pNext;
-          }
           result = getDispatcher()->vkGetEncodedVideoSessionParametersKHR(
             static_cast<VkDevice>( m_device ),
             reinterpret_cast<const VkVideoEncodeSessionParametersGetInfoKHR *>( &videoSessionParametersInfo ),

